junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/TempDirectoryPerDeclarationTests.java

```diff
@@ -1131,4 +1150,46 @@ public Path createTempDirectory(ExtensionContext context) throws Exception {
 
 	}
 
+	static class FactoryWithMemoryFileSystemTestCase {
+
+		@Test
+		void test(@TempDir(factory = Factory.class) Path tempDir) {
+			assertThat(tempDir).exists().hasFileSystem(Factory.fileSystem);
+			assertThat(tempDir.getFileName().toString()).startsWith("prefix");
+		}
+
+		private static class Factory implements TempDirFactory {
+
+			private static FileSystem fileSystem;
+
+			@Override
+			public Path createTempDirectory(ExtensionContext context) throws Exception {
+				fileSystem = MemoryFileSystemBuilder.newEmpty().build();
+				return Files.createTempDirectory(fileSystem.getPath("/"), "prefix");
+			}
+		}
+
+	}
+
+	static class FactoryWitJimfsTestCase {
+
+		@Test
+		void test(@TempDir(factory = Factory.class) Path tempDir) {
```

Comment:
What should happen in the following case:
```java
@Test
void test(@TempDir(factory = Factory.class) Path a, @TempDir(factory = Factory.class) Path b)
```

Without specifying `factory`, two different paths from the same (default) file system will be injected. With `factory` they will be from different file systems. I wonder if we need to support that use case somehow. 🤔

Reply:
Not sure I understood your concern.

In case the same factory is specified on both annotations, how the underlying file system is handled should be an implementation detail of the factory.

Maybe you're referring to the fact that each annotation will instantiate a different factory. If something like the file system has to be shared, it should be declared as static (with additional care in the `close()` method).

Is that what you were thinking?

The Javadoc could maybe highlight such cases. What do you think?

Reply:
Yes, I was wondering if there's going to be a need to use the same `FileSystem`. However, one could even solve this by using the `ExtensionContext` and putting a `CloseableResource` in the root `Store`. For now, I don't think we have to document anything in particular given we have no real-world use case.